### DISCUSSION TITLE
Resolution of non-literal `.` when `README.md` is in a parent directory. was: ~Environment location~

### DISCUSSION THREAD

**Discussion by jhheider** [pkgxdev/pkgx#270] [Ideas]
Edit: the README.md is being properly located; this has become about my counter-intuitive internal mental translation of path `.` to the `getting-started` script. I should probably stop doing that.

Given how any number of tools work (`npm` and `just` spring immediately to mind), what is less surprising here:
```sh
$ tea -X git clone https://github.com/teaxyz/cli
$ cd cli
$ tea .
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   105  100   105    0     0    497      0 --:--:-- --:--:-- --:--:--   507
100 74.4M  100 74.4M    0     0  26.1M      0  0:00:02  0:00:02 --:--:-- 31.8M
tea: installed: ~/.tea/charm.sh/glow/v1.4.1

   tea really is a standalone binary  
$ cd src
$ tea .
### what do you expect here?
```

Current behavior is to report `tea: command not found: undefined`. Tools, such as those mentioned at the top, would instead start walking up the tree looking for `README.md`, and locating it at `..`. However, if we do _that_, should $CWD be `..` or `.` during execution? Do we need a $SUPER_PWD within scripts, to allow for both behaviors to be available?

I'm not sure there's One True Answer to this, but I wonder what's most/least surprising to users.

**Comment by jonchang** [comment]
I imagine it would walk up the parent directory chain, stopping at filesystem mounts, looking for README.md. CWD should be the current directory. Basically, how `git status` works, i.e., stopping at the first `.git` directory found or a system mount, with the paths returned by git status (in the case of a dirty work tree) relative to the current directory.

**Comment by mxcl** [comment]
So… we already do walk up the directory tree. We stop at `$HOME` and `/` (same behavior as `git`).

Is something not working as expected?

**Reply by jhheider** [reply]
Hm. I seem to be getting exe/md not found for `tea .`. Which makes a kind of sense, since that's not the correct path to the readme. We walk for the venv, right?

**Reply by mxcl** [reply]
we walk up looking for a readme yes. I'm assuming it does not find one and then hits `$HOME`. Is that not what is happening?

**Reply by jhheider** [reply]
This is what I mean (using teaxyz/cli):
```sh
 ~/tea/cli $ tea .
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   105  100   105    0     0    933      0 --:--:-- --:--:-- --:--:--  1009
100 74.9M  100 74.9M    0     0  46.2M      0  0:00:01  0:00:01 --:--:-- 57.8M
tea: installed: ~/.tea/charm.sh/glow/v1.4.1
^C
 ~/tea/cli $ cd src
 ~/tea/cli/src $ tea .
tea: command not found: undefined
 ~/tea/cli/src $ tea ..
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
^C
 ~/tea/cli/src $ tea getting-started
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
^C
```

So, it's really only `tea .`, which is magical shorthand, that "doesn't work" in subdirectories. In my head, I've been translating as `alias .=getting-started`, but that is a different kind of confusing probably. Taken as a literal path, it makes perfect sense as implemented.

**Reply by mxcl** [reply]
K, so probs this is a bug RN, but I do think we should rethink how exe/md targets are invoked and that.
